File: ios/Sources/ACCore/Models/Elements/TabSet.swift
import Foundation
// MARK: - TabSet

public struct TabSet: Codable, Equatable {
    public let type: String = "TabSet"
    public var id: String?
    public var tabs: [Tab]
    public var selectedTabId: String?
    public var spacing: Spacing?
    public var separator: Bool?
    public var height: BlockElementHeight?
    public var isVisible: Bool?
    public var requires: [String: String]?

    public init(
        id: String? = nil,
        tabs: [Tab],
        selectedTabId: String? = nil,
        spacing: Spacing? = nil,
        separator: Bool? = nil,
        height: BlockElementHeight? = nil,
        isVisible: Bool? = nil,
        requires: [String: String]? = nil
    ) {
        self.id = id
        self.tabs = tabs
        self.selectedTabId = selectedTabId
        self.spacing = spacing
        self.separator = separator
        self.height = height
        self.isVisible = isVisible
        self.requires = requires
    }
}

public struct Tab: Codable, Equatable {
    public var id: String
    public var title: String
    public var icon: String?
    public var items: [CardElement]

    public init(
        id: String,
        title: String,
        icon: String? = nil,
        items: [CardElement]
    ) {
        self.id = id
        self.title = title
        self.icon = icon
        self.items = items
    }
}

File: android/ac-core/src/main/kotlin/com/microsoft/adaptivecards/core/models/TabSet.kt
package com.microsoft.adaptivecards.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import kotlinx.serialization.json.JsonElement

@Serializable
@SerialName("TabSet")
data class TabSet(
    @Transient override val type: String = "TabSet",
    override val id: String? = null,
    override val isVisible: Boolean = true,
    override val separator: Boolean = false,
    override val spacing: Spacing? = null,
    override val height: BlockElementHeight? = null,
    override val requires: Map<String, String>? = null,
    override val fallback: JsonElement? = null,
    val tabs: List<Tab>,
    val selectedTabId: String? = null
) : CardElement

@Serializable
data class Tab(
    val id: String,
    val title: String,
    val icon: String? = null,
    val items: List<CardElement>
)
